src/material/sidenav/drawer.ts

```diff
@@ -809,19 +809,21 @@ export class MatDrawerContainer implements AfterContentInit, DoCheck, OnDestroy
     this._start = this._end = null;
 
     // Ensure that we have at most one start and one end drawer.
-    this._drawers.forEach(drawer => {
-      if (drawer.position == 'end') {
-        if (this._end != null) {
-          throwMatDuplicatedDrawerError('end');
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
```

Comment:
@crisbeto just noticed this one is actually gating actual behavior for the component (setting `this._end` or `this._start`)

Reply:
ops, good catch. Fixed, although it does reveal that if we have some faulty logic, our tests won't catch it since they're only being run in dev mode.